app/uk/gov/hmrc/preferenceschangednotifier/connectors/EpsHodsAdapterConnector.scala

```diff
@@ -19,37 +19,48 @@ package uk.gov.hmrc.preferenceschangednotifier.connectors
 import com.google.inject.Inject
 import play.api.Logging
 import uk.gov.hmrc.http.HttpReads.Implicits._
-import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse}
-import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
-import uk.gov.hmrc.preferenceschangednotifier.model.{
-  NotifySubscriberRequest,
-  Subscriber
+import uk.gov.hmrc.http.{
+  HeaderCarrier,
+  HttpClient,
+  HttpResponse,
+  UpstreamErrorResponse
 }
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+import uk.gov.hmrc.preferenceschangednotifier.model.NotifySubscriberRequest
 
 import java.util.UUID
+import javax.inject.Singleton
 import scala.concurrent.{ExecutionContext, Future}
 
-class EpsHodsAdapterConnector @Inject()(http: HttpClient,
-                                        servicesConfig: ServicesConfig)(
-    implicit ec: ExecutionContext,
-    hc: HeaderCarrier)
+@Singleton
+class EpsHodsAdapterConnector @Inject()(
+    http: HttpClient,
+    servicesConfig: ServicesConfig)(implicit ec: ExecutionContext)
     extends Subscriber
     with Logging {
 
   lazy val baseUrl: String = servicesConfig.baseUrl("eps-hods-adapter")
 
-  private def postUrl = s"$baseUrl/eps-hods-adapter/notify-subscriber"
+  private def postUrl =
+    s"$baseUrl/eps-hods-adapter/preferences/notify-subscriber"
 
   override def notifySubscriber(
-      request: NotifySubscriberRequest): Future[HttpResponse] = {
+      request: NotifySubscriberRequest
+  )(implicit hc: HeaderCarrier)
+    : Future[Either[UpstreamErrorResponse, HttpResponse]] = {
 
-    logger.debug(s"notify-subscriber")
-
-    http
-      .POST[NotifySubscriberRequest, HttpResponse](
+    val aaa = http
```

Comment:
May be this is not intentional to assign this to val

Reply:
Yes, you are right, was debugging something and did not set it back - thanks!